packages/playwright-core/src/server/android/android.ts

```diff
@@ -343,6 +345,35 @@ export class AndroidDevice extends SdkObject {
     validateBrowserContextOptions(options, browserOptions);
 
     const browser = await CRBrowser.connect(this.attribution.playwright, androidBrowser, browserOptions);
+    {
+      const androidArtifacts = new Set<string>();
+      browser._interceptSetInputFilePaths = async (page, files) => {
```

Comment:
This is quite a hack. I'd rather get createTempFile to create files on the android device, wdyt?